src/server.c

```diff
@@ -7500,9 +7491,6 @@ static void sigShutdownHandler(int sig) {
         serverLogFromHandler(LL_WARNING, "You insist... exiting now.");
         rdbRemoveTempFile(getpid(), 1);
         exit(1); /* Exit with an error since this was not a clean shutdown. */
-    } else if (server.loading) {
-        serverLogFromHandler(LL_WARNING, "Received shutdown signal during loading, exiting now.");
-        exit(0);
```

Comment:
let's mention this change in the top comment

Reply:
I'm reverting this cleanup, because there are some failing test cases that I'd love to avoid spending hours debugging... See the commit message with the revert.

Reply:
ok, i don't understand why it fails, and it's only one test, so probably easy to solve, but we can circle back to it some other time.
p.s. looking at the reverted commit again, it seems you didn't set the NOSAVE flag in that flow.

Reply:
SHUTDOEN_NOSAVE is set in prepareForShutdown if `server.loading`. This is already in unstable.

Reply:
i now realized that serverCron doesn't run during loading.
we send SIGTERM to the replica to ask that it'll die, but it ignores it and keep on reading the rdb file.

i think the solution is that the signal handler should call `prepareForShutdown(FORCE|NOW|NOSAVE) && exit(0)` instead of just `exit(0)`

Reply:
For reference: In a chat, we decided to postpone this to another PR.

The signal handler should only touch variables declared `volatile sig_atomic_t` or similar, like `shutdown_asap`.

Another option is to call prepareForShutdown from whileBlockedCron, but the test cases still fail, possibly because they don't write enough data to trigger whileBlockedCron. Maybe it's enough to tweak the test cases a bit.

Reply:
I asked @perryitay to take a look at it (he's doing something else in that area)